services/src/main/java/org/fao/geonet/api/regions/metadata/MetadataRegionSearchRequest.java

```diff
@@ -168,33 +193,75 @@ private void loadAll(List<Region> regions, Id id) throws Exception {
     }
 
     Region parseRegion(Id mdId, Element extentObj) throws Exception {
+        MultiPolygon geometry = parseElement(extentObj);
+
+        String id = null;
+        if (geometry != null) {
+            Element element = extentObj.getChild("element", Geonet.Namespaces.GEONET);
+            if (element != null) {
+                id = element.getAttributeValue("ref");
+            }
+            return new MetadataRegion(mdId, id, geometry);
+        } else {
+            return null;
+        }
+    }
+
+    private MultiPolygon parseElement(Element extentObj) throws Exception {
         GeonetContext gc = (GeonetContext) context.getHandlerContext(Geonet.CONTEXT_NAME);
         gc.getBean(DataManager.class).getEditLib().removeEditingInfo(extentObj);
 
-        String id = null;
-        Geometry geometry = null;
+        MultiPolygon geometry = null;
         if ("polygon".equals(extentObj.getName())) {
-            String gml = Xml.getString(extentObj);
-            geometry = SpatialIndexWriter.parseGml(parsers[0], gml);
+            geometry = parsePolygon(extentObj);
         } else if ("EX_BoundingPolygon".equals(extentObj.getName())) {
-            String gml = Xml.getString(extentObj.getChild("polygon", Geonet.Namespaces.GMD));
-            geometry = SpatialIndexWriter.parseGml(parsers[0], gml);
+            Element polygon = extentObj.getChild("polygon", Namespaces.GMD);
```

Comment:
Should we add ISO19115-3 case here too ? ie `gex:polygon` and `gex:westBoundLongitude`

Reply:
Yes. looking at 19115-3 now

Reply:
Support for 19115-3 added.  PR for 19115-3:2018 to follow